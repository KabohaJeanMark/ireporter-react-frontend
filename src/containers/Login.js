import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import LoginPresentationComponent from '../components/LoginComponent';
import PropTypes from 'prop-types';
import {loginUrl} from '../utils';
import {postLogin} from '../redux/actions/LoginActions';


export class Login extends Component {
  constructor(props){
    super(props);
    const state = {
      username:"",
      password:""
    };
  }

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };
  handleSubmit = (event) => {
    event.preventDefault();
    const { username,  password } = this.state;
    const loginDetails = { username,  password };
    this.props.postLogin(loginUrl, loginDetails, this.props);
  };

  render () {
    return (
      <div>
      <LoginPresentationComponent
        handleSubmit={this.handleSubmit}
        handleChange={this.handleChange}/>
      </div>
      
    );
  }
}

Login.propTypes = {
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func,
  first_name: PropTypes.string,
  last_name: PropTypes.string,
  username: PropTypes.string,
  othernames: PropTypes.string,
  phone_number:PropTypes.string,
  email:PropTypes.string,
  password:PropTypes.string,
  postLogin: PropTypes.func
};

export default connect(null, {postLogin})(Login);
