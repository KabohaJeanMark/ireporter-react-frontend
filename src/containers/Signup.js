import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import SignUpPresentationComponent from '../components/SignupComponent';
import PropTypes from 'prop-types';
import {signupUrl} from '../utils';
import {postSignup} from '../redux/actions/SignupActions';


export class SignUp extends Component {
  constructor(props){
    super(props);
    const state = {
      first_name:"",
      last_name:"",
      othernames:"",
      username:"",
      phone_number:"",
      email:"",
      password:""
    };
  }
  checkBlank = () => {
    const { first_name, last_name, othernames, username, phone_number,email, password } = this.state;
    if (first_name === '' || last_name === '' || othernames === ''||username === '' || phone_number === '' || email === ''|| password ==='') {
      toast.dismiss();
      toast.error("All fields must be filled", {
        hideProgressBar: false,
        autoClose: 3000
      });
    }
  };

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };
  handleSubmit = (event) => {
    event.preventDefault();
    const { first_name, last_name, othernames, username, phone_number,email, password } = this.state;
    const signupDetails = { first_name, last_name, othernames, username, phone_number,email, password };
    this.checkBlank();
    this.props.postSignup(signupUrl, signupDetails, this.props);
  };

  render () {
    return (
      <div>
      <SignUpPresentationComponent
        handleSubmit={this.handleSubmit}
        handleChange={this.handleChange}/>
      </div>
      
    );
  }
}

SignUp.propTypes = {
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func,
  first_name: PropTypes.string,
  last_name: PropTypes.string,
  username: PropTypes.string,
  othernames: PropTypes.string,
  phone_number:PropTypes.string,
  email:PropTypes.string,
  password:PropTypes.string,
  postSignup: PropTypes.func
};

export default connect(null, {postSignup})(SignUp);
