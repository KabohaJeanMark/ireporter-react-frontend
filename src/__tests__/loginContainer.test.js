import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Login } from '../containers/Login';

describe('Login container', () => {
  let wrapper;
  let props;
  let instance;
  beforeEach(() => {
    props = {
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      postLogin: jest.fn(),
      username: "marcus",
      password: "kakashi@1994",
      toast: {
        error: jest.fn()
      }
    };
    wrapper = shallow(<Login {...props} />);
    instance = wrapper.instance();
  });
  it('renders without crashing', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('handles form submission', () => {
    const e = {
      preventDefault: jest.fn()
    };
    instance.setState(props);
    instance.handleSubmit(e);
    expect(props.postLogin).toHaveBeenCalledTimes(1);
  });
  it('handles change', () => {
    const e = {
      target: {
        name: "username",
        value: "marcusrashford"
      }
    };
    instance.handleChange(e);
  });
});
