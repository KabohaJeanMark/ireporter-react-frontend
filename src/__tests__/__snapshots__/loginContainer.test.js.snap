// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login container renders without crashing 1`] = `
<div>
  <LoginPresentationComponent
    handleChange={[Function]}
    handleSubmit={[Function]}
  />
</div>
`;
